Include/cpython/pystate.h

```diff
@@ -191,6 +191,8 @@ struct _ts {
     PyObject *previous_executor;
 
     uint64_t dict_global_version;
+    int sp_cached; /* Only used in debug builds */
```

Comment:
If only used in debug builds, should this be inside `#ifdef Py_DEBUG`?

Reply:
I *think* debug builds have the same ABI compatibility as release builds these days. I know we like to pretend that `PyThreadState` is opaque, but this seems like it *might* case some nasty bugs if we do this.